Vision_feats/generate_feats.py

```diff
@@ -0,0 +1,308 @@
+from ast import Raise
+from builtins import breakpoint
+import glob
+import os
+import csv
+import clip
+import argparse
+from pyexpat import model
+from turtle import forward
+import pandas as pd
+import numpy as np
+from PIL import Image
+from tqdm import tqdm
+from typing import Dict, List, Tuple
+
+import torch
+import torch.nn as nn
+import torch.utils.data as data
+from torch.utils.data import DataLoader
+
+import torchvision
+import torchvision.transforms as transforms
+from torchvision.models import resnet50
+from transformers import BeitFeatureExtractor, BeitModel
+
+models_file = {
+    'beit': "microsoft/beit-large-patch16-224-pt22k",
+    'beit_ft': "microsoft/beit-large-patch16-224-pt22k-ft22k",
+    'clip': "ViT-B/32",
+    'clipL': "ViT-L/14@336px" 
+}
+
+
+class ImageLoader(data.Dataset):
+    """The bare minimum class for 
+        loading image data
+        inference purpose only
+    """
+    def __init__(
+        self,
+        args,
+        img_name_file=None,
+    ):
+        # # TODO: Fix the place of feature extractor
+        BeitFeat = BeitFeatureExtractor.from_pretrained(
+            "microsoft/beit-large-patch16-224-pt22k"
+        )
+        BeitFtFeat = BeitFeatureExtractor.from_pretrained(
+            "microsoft/beit-large-patch16-224-pt22k-ft22k"
+        )
+        _, ClipFeat = clip.load("ViT-B/32")
+        _, ClipFeatL = clip.load("ViT-L/14@336px")
+        self.transforms_dict = {
+            'resnet':
+                transforms.Compose(
+                    [
+                        transforms.Resize(256),
+                        transforms.CenterCrop(224),
+                        transforms.ToTensor(),
+                        transforms.Normalize(
+                            mean=[0.485, 0.456, 0.406],
+                            std=[0.229, 0.224, 0.225]
+                        ),  #Imagenet pretraining default normalization.
+                    ]
+                ),
```

Comment:
Two things here: (1) is there any way to not hard code these things for ResNet? No worries if not.  (2) this dict thing feels ad-hoc.  Maybe make a super-class with a transforms object and all of the other logic, and then subclasses for ResNet / BeIT, that basically only set the transforms to the proper thing, since that's the difference

Reply:
(1) Good point, though I think it makes sense to hardcode these because ResNet was kind of trained on these preprocessing steps and we don't want future users to change that?
(2) Yeah, I was thinking of doing that before, somehow the transform/preprocessing step has to happen in the image loading stage since they are essential in turning the img into well-formatted tensors. (I might misunderstand what you mean here tho)